src/Microsoft.AspNetCore.Mvc.Core/ApiBehaviorOptions.cs

```diff
@@ -52,5 +70,43 @@ public class ApiBehaviorOptions
         /// that are bound from form data.
         /// </summary>
         public bool SuppressConsumesConstraintForFormFileParameters { get; set; }
+
+
+        public bool AllowUseProblemDetailsForClientErrorResponses
+        {
+            get => _allowUseProblemDetailsForClientErrorResponses.Value;
+            set => _allowUseProblemDetailsForClientErrorResponses.Value = value;
+        }
+
+        public IDictionary<int, Func<ActionContext, ProblemDetails>> ProblemDetailsFactory { get; } = new Dictionary<int, Func<ActionContext, ProblemDetails>>
```

Comment:
Or alternatively, make this a service that calls into options. Or both.

I can see people having opinions about how this should behave.

Reply:
Actually I think we can skip the service since we didn't do it for model state error factory (just options).

Reply:
If we were to let it return arbitrary types (which I'm 🆗 with since we do that for `InvalidModelStateResponseFactory`), this would be `Func<ActionContext, IActionResult>` - you typically also would want to set the content-type on the result.